src/modules/nats/nats.service.ts

```diff
@@ -1,28 +1,60 @@
-import { Injectable, OnModuleDestroy } from '@nestjs/common';
-import { Client, connect } from 'nats';
+import { Injectable } from '@nestjs/common';
+import { InjectQueue, Process, Processor } from '@nestjs/bull';
+import { Job, Queue } from 'bull';
+import { NatsGateway } from './nats.gateway';
 import { ConfigService } from '@nestjs/config';
 import { Logger } from '../logger/logger.service';
 
+export type IMessageJob = {
+  subject: string;
+  data: Record<string, unknown>;
+};
 @Injectable()
-export class NatsService implements OnModuleDestroy {
-  public connection: Client;
-  constructor(private config: ConfigService, private readonly logger: Logger) {
-    const NATS_CLIENTS_URL = this.config.get<string>('NATS_CLIENTS_URL');
-    try {
-      this.connection = connect(`nats://${NATS_CLIENTS_URL}`, {
-        pingInterval: 5000,
-        maxReconnectAttempts: -1,
-        reconnectDelayHandler: () => {
-          this.logger.log(`Reconnecting to NATS server...`);
-          return 1000;
-        },
-      });
-    } catch (err) {
-      this.logger.error(err, NatsService.name);
-    }
+@Processor('nats-messages')
+export class NatsService {
+  natsSubjectSuffix: string;
+
+  constructor(
+    private natsGateway: NatsGateway,
+    private logger: Logger,
+    private readonly config: ConfigService,
+    @InjectQueue('nats-messages')
+    private readonly messagesQueue: Queue<IMessageJob>,
+  ) {
+    this.natsSubjectSuffix = this.config.get<string>('NATS_SUBJECT_SUFFIX');
+  }
+
+  public async publish(subject: string, data: Record<string, unknown>) {
```

Comment:
Is this used outside of class? Can we make it `private`?
```suggestion
  private async publish(subject: string, data: Record<string, unknown>) {
```

Reply:
It's in use in one place , I'll confirm if we can turn it into publishDids and hide this one.